File: acm/10038.pas
program JollyJumpers;
var
	a : array[0..3000] of Boolean;
	m, n, t  : longint;
	i, j, k : longint;
begin
	while not eof do
	begin
		fillchar(a,sizeof(a),0);
		read(k);
		read(n);
		if (k = 1) then
		begin
			writeln('Jolly');
			readln;
			continue;
		end;
		for i := 2 to k do
		begin
			m := n;
			read(n);
			t := abs(m - n);
			if (t = 0) or (t > k-1) or (a[t mod 3000]) then
			begin
				writeln('Not jolly');
				for j := i + 1 to k do
					read(n);
				readln;
				break;
			end;
			a[t] := TRUE;
		end;
		if (i = k) then
			writeln('Jolly');
	end;
end.

File: acm/10038.c
#include <stdio.h>
#include <math.h>
#include <string.h>

#define false 0
#define true 1

int main() {
	int x, y, z, n;
	int i,j,k;
	int a[4000];

	while ( scanf("%d", &n) != EOF ) {
		memset(a,0,sizeof(a));
		scanf("%d", &x);
		for (i=0;i<n-1;i++) {
			scanf("%d", &y);
			z = abs(x-y);
			if (z <= 3000) { a[z]++; };
			x = y;
		};
	/*	for (i=0;i<n;i++) {
			printf("%d ", a[i]);
		};printf("\n");*/
		z = false;
		for (i=1;i<n;i++) {
			if (a[i] != 1) { printf("Not jolly\n"); z = true; break; };
		};
		if (z ==false ){ printf("Jolly\n");};
		};
	return 0;
};

